### PR TITLE
Account deletion with code confirmation endpoints

### PR DESCRIPTION
**Pull request by BartkiewiczM**
I've implemented two endpoints:
/api/user/delete_account_send_code
This endpoint saves a confirmation code for a logged-in user in the database. The code is valid for 30 minutes.
It sends a email to this user with the code for confirmation in the app.

/api/user/delete_account_confirm
This endpoint takes the code that is sent from the app and matches it with the one that's saved in the database.
If the code passes the validation the account is deleted.

### CODE REVIEW

File: app/controllers/api/delete_account_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+module Api
+  class DeleteAccountController < BaseController
```
Comment (mpanuszewska) [inline app/controllers/api/delete_account_controller.rb:2]: :)

File: spec/controllers/api/delete_account_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+      it 'successfully deletes the user account' do
+        expect { post :delete_account_with_confirmation_code, params: { code: '1234' }, format: :json }
+        .to change(User, :count).by(-1)
+        expect(response).to have_http_status(:ok)
+        expect(json_response['message']).to eq('Account successfully deleted.')
+      end
+    end
+
+    context 'with an invalid or expired confirmation code' do
```
Comment (mpanuszewska) [inline spec/controllers/api/delete_account_controller_spec.rb:33]: This is not really testing expired (but valid) confirmation codes, I'd probably add that too. Dates are always tricky.

**Review by mpanuszewska — APPROVED** [review]
Looks really good! 👏 

I've been wondering if we should cleanup old codes after they expire but couldn't really find a good reason so it's probably not reasonable. 

One small comment on the tests, otherwise you're good to go!

**Comment by BartkiewiczM** [comment]
Thank you! I think for now the old endpoint will be used by people before they update the Android/iOS app.

**Pull request merged**